lib/exe-type.js

```diff
@@ -23,9 +23,29 @@ function isSh(file) {
   return ['dash', 'sh', 'bash', 'zsh'].includes(file)
 }
 
+/**
+ * Returns the basename of the executable.
+ *
+ * On Windows, strips the `.exe` extension (if any) and normalizes the name to
+ * lowercase.
+ *
+ * @param exePath {string} Path of the executable as passed to spawned processes:
+ *   either command or a path to a file.
+ * @return {string} Basename of the executable.
+ */
+function getExeBasename(exePath) {
+  const baseName = path.basename(exePath);
+  if (isWindows()) {
+    return baseName.replace(/\.exe$/i, "").toLowerCase();
```

Comment:
Should this also strip `.cmd` if found?  Probably no because it looks like this is replacing code that did not strip `.cmd` but I figure I should ask anyways.

Reply:
I don't know. I tend to think that stripping `.bat` and `.cmd` would be a good thing to normalize the executable name across platforms. If I remember correctly, there is a way to define executable extensions in Windows (either in the registry or using an environment variable).

I'll add a comment in the code about this but leave it as is for the moment.